### PR TITLE
Sahilsharma356/ch26/redo stopwatch lap UI

### PR DESCRIPTION
**Pull request by SVD6**
Story: https://app.clubhouse.io/chrono/story/26/redo-stopwatch-lap-ui

Known Issues: If you make enough laps such that you have to scroll, the first lap will be a little blurred out because of the fade effect that we have going. Gonna have to look into a fix for that but it's not that important right now.

Needs UI review from Aarthi but I think it looks pretty good! Lap functionality works, reset button does what it's supposed to, etc.

### CODE REVIEW

File: app/src/main/java/com/example/chrono/main/stopwatch/StopwatchFrag.kt [hunk 1]
```diff
@@ -65,72 +68,46 @@ class StopwatchFrag : Fragment() {
...
+    private fun lap() {
+        lapCount += 1
+        val currTime = SystemClock.elapsedRealtime() - chronometer!!.base
+        val timeDiff = currTime - prevTime
+
+        // Create Lap Object
+        val lap = LapObject()
+        lap.lapNum = lapCount
+        lap.lapTime = timeDiff
+        lap.totalTime = SystemClock.elapsedRealtime() - chronometer!!.base
```
Comment (harigovind1998) [inline app/src/main/java/com/example/chrono/main/stopwatch/StopwatchFrag.kt]: use currTime

File: app/src/main/java/com/example/chrono/util/UtilClass.kt [hunk 2]
```diff
@@ -32,6 +33,36 @@ fun getIconName(position: Int): String {
     return "-1"
 }
 
+fun getTime(timeElapsed: Long): String {
```
Comment (harigovind1998) [inline app/src/main/java/com/example/chrono/util/UtilClass.kt:36]: The same calculate is done in MyChronometer.updateText(). Make MyChronometer.updateText() call this function instead. Can be done as part of the code clean up story.

File: app/src/main/res/layout/fragment_stopwatch.xml [hunk 3]
```diff
@@ -6,23 +6,24 @@
...
             android:layout_width="match_parent"
-            android:layout_height="2in">
+            android:layout_height="match_parent"
+            android:layout_above="@+id/chrono_layout"
+            app:fel_edge="top"
+            app:fel_size_top="@dimen/huge_element_padding">
 
-            <ScrollView
+            <androidx.recyclerview.widget.RecyclerView
+                android:id="@+id/recyclerView"
```
Comment (harigovind1998) [inline app/src/main/res/layout/fragment_stopwatch.xml]: recycler_view

File: app/src/main/res/layout/lap_lay.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:gravity="center"
+    android:orientation="horizontal"
+    android:padding="@dimen/tiny_element_padding">
+
+    <com.google.android.material.textview.MaterialTextView
+        android:id="@+id/lapNum"
```
Comment (harigovind1998) [inline app/src/main/res/layout/lap_lay.xml]: lap_num

File: app/src/main/res/layout/lap_lay.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_vertical"
+        android:layout_marginHorizontal="@dimen/small_element_padding"
+        android:textAllCaps="false"
+        android:textColor="@color/lap_blue"
+        android:textSize="@dimen/small_text"
+        android:textStyle="bold" />
+
+    <com.google.android.material.textview.MaterialTextView
+        android:id="@+id/lapTime"
```
Comment (harigovind1998) [inline app/src/main/res/layout/lap_lay.xml]: lap_time

File: app/src/main/res/layout/lap_lay.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_vertical"
+        android:layout_marginHorizontal="@dimen/small_element_padding"
+        android:textAllCaps="false"
+        android:textSize="@dimen/small_text"
+        android:textStyle="bold" />
+
+    <com.google.android.material.textview.MaterialTextView
+        android:id="@+id/totalTime"
```
Comment (harigovind1998) [inline app/src/main/res/layout/lap_lay.xml]: total_time

File: app/src/main/java/com/example/chrono/main/stopwatch/StopwatchFrag.kt (hunk 1 above)
Comment (SVD6) [reply app/src/main/java/com/example/chrono/main/stopwatch/StopwatchFrag.kt]: 👍

File: app/src/main/java/com/example/chrono/util/UtilClass.kt (hunk 2 above)
Comment (SVD6) [reply app/src/main/java/com/example/chrono/util/UtilClass.kt:36]: Yeah so I kinda fucked up cuz I did some refactoring in the clean up story which I shouldn't have until this was merged in as well. It makes sense the way I did it in that story so I'm gonna merge this in there and work it out in that story. Merge conflicts here I come :(

File: app/src/main/res/layout/fragment_stopwatch.xml (hunk 3 above)
Comment (SVD6) [reply app/src/main/res/layout/fragment_stopwatch.xml]: 👍

File: app/src/main/res/layout/lap_lay.xml (hunk 5 above)
Comment (SVD6) [reply app/src/main/res/layout/lap_lay.xml]: 👍

File: app/src/main/res/layout/lap_lay.xml (hunk 6 above)
Comment (SVD6) [reply app/src/main/res/layout/lap_lay.xml]: 👍

File: app/src/main/res/layout/lap_lay.xml (hunk 4 above)
Comment (SVD6) [reply app/src/main/res/layout/lap_lay.xml]: 👍

**Pull request merged**